build/cmake/DefineOptions.cmake

```diff
@@ -62,8 +62,10 @@ endif()
 find_package(OpenSSL QUIET)
 CMAKE_DEPENDENT_OPTION(WITH_OPENSSL "Build with OpenSSL support" ON
                        "OPENSSL_FOUND" OFF)
-option(WITH_BOOSTTHREADS "Build with Boost thread support" OFF)
 option(WITH_STDTHREADS "Build with C++ std::thread support" OFF)
+CMAKE_DEPENDENT_OPTION(WITH_BOOSTTHREADS "Build with Boost threads support" ON
```

Comment:
@nsuke  Well, then we can change in following way:

CMAKE_DEPENDENT_OPTION(WITH_BOOSTTHREADS "Build with Boost threads support" ON
-    "NOT WITH_STDTHREADS;Boost_FOUND" OFF)
+    "NOT WITH_STDTHREADS;Boost_FOUND;WITH_BOOSTTHREADS" OFF)

Then, this feature will not be enabled if not explicitly set to on using -DWITH_BOOSTTHREADS=ON in command line.

What do you think?

BTW, I've found that this option will set add_definitions, but not update config.h
Is it normal?